daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/InfosetOutputter.scala

```diff
@@ -54,11 +54,12 @@ trait InfosetOutputter {
    * @param diSimple the simple element that is started. Various fields of
    *                 DISimple can be accessed to determine things like the
    *                 value, nil, name, namespace, etc.
+   * @param xmlOutputStyle the string that determines how to print data.
    * @return true on sucess, false if there was an error and Daffodil should stop all
    *         future calls to the InfosetOutputter
    */
 
-  def startSimple(diSimple: DISimple): Boolean
+  def startSimple(diSimple: DISimple, xmlOutputStyle: String): Boolean
```

Comment:
I'm not sure I like this change to the `InfosetOutputter`.

For one, it breaks API backwards compatibility, which isn't a huge deal since we don't know of other implementations outside of Daffodil, but it's still not great.

Though, my main gripe with this is that this CDATA stuff is so specific to the `XMLTextInfosetOutputter`. The `JSONInfosetOutputter` definitely doesn't need it, and other outputters create actual java objects, and so whitespace isn't really an issue for them unless they are pretty printed. But in that case the pretty printers can figure out this logic which is outside the scope of Daffodil.

Because this is specific to the `XMLTextInfosetOutputter`, I would suggest that this should instead be a parameter passed to its constructor, and it's up to the caller how to figure out how that gets in there. For API users that's easy since they can just supply the new parameter. For CLI users, it's a bit more tricky since we don't yet have a way for the CLI to provide options to infoset outputters. This is a similar issue as [DAFFODIL-2234](https://issues.apache.org/jira/browse/DAFFODIL-2234), so maybe we figure that out separately, and this is only a feature for API users until then?

This would also man that you don't have to pass this xmlOutputStyle around all over the place just to get it to the one outputter that actually cares about it. It should reduce the changes quite a bit.

Reply:
@stevedlawrence can this conversation be marked as resolved?